Lib/fontTools/ttLib/tables/otTables.py

```diff
@@ -562,7 +562,7 @@ def preWrite(self, font):
 
 		glyphOrder = font.getGlyphOrder()
 		mapping = [mapping[g] for g in glyphOrder]
-		while mapping and mapping[-2] == mapping[-1]:
+		while mapping and len(mapping) > 1 and mapping[-2] == mapping[-1]:
```

Comment:
The check for "mapping" itself is redundant now. So, please rewrite as "while len(mapping) > 1 and ..."

Reply:
Done.